src/common/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2017-2024 Smart code 203358507
+
+import React, { useState, useEffect, useCallback, DetailedHTMLProps, HTMLAttributes } from 'react';
+import classNames from 'classnames';
+import styles from './Checkbox.less';
+import Icon from '@stremio/stremio-icons/react';
+
+type Props = {
+    disabled?: boolean;
+    value?: boolean;
+    className?: string;
+    onChange?: (checked: boolean) => void;
+    ariaLabel?: string;
+    error?: string;
+};
+
+const Checkbox = ({ disabled, value, className, onChange, ariaLabel, error }: Props) => {
+    const [isChecked, setIsChecked] = useState(false);
+    const [isError, setIsError] = useState(false);
+    const [isDisabled, setIsDisabled] = useState(disabled);
+
+    const handleChangeCheckbox = useCallback(() => {
```

Comment:
Use the native event for the checkbox state
```suggestion
    const handleChangeCheckbox = useCallback(({ target }: ChangeEvent<HTMLInputElement>) => {
        onChange && onChange(target.checked);
    }, [onChange]);
```